Refactor microk8s playbook to follow "new" structure

Chickensoupwithrice opened the pull request:
- Integrates with btrix/deploy role for deploying
- Seperated RedHat and Debian into seperate roles
- Created Common role

Tested on DO Debian 12 box with the following command:

```
ansible-playbook -i inventory/microk8s/hosts.ini install_microk8s.yml -e host_ip="127.0.0.1" -e domain_name="test.browsertrix.cloud" -e your_user="root"
```

Chickensoupwithrice committed:
Refactor microk8s playbook to follow "new" structure

- Integrates with btrix/deploy role for deploying
- Seperated RedHat and Debian into seperate roles
- Created Common role

Tested on DO Debian 12 box

ikreymer commented on ansible/inventory/microk8s/group_vars/main.yml:
```diff
@@ -0,0 +1,38 @@
+
+your_user: root          # if your cloud provide uses another user change this
+browsertrix_cloud_home: "/opt/browsertrix-cloud"
+microk8s_user_home: "/home/{{ microk8s_user }}"
```
should this be `ansible_user` to match the changes in the other main.yml?

Chickensoupwithrice committed:
fix(ansible):remove unused values / templates files

Chickensoupwithrice committed:
feat(ansible): add starling changes

Chickensoupwithrice committed:
fix(ansible): remove old playbook

Chickensoupwithrice committed:
feat(ansible): add starling exmaple hosts file

Chickensoupwithrice committed:
fix(ansible): remove references to starling

ikreymer committed:
additional fixes:
- allow running remotely by default
- use 'browsertrix_cloud_home' for charts path
- add additional customizable options to btrix_values.j2 (todo: unify all the templates)
- docs: update to new playbook path
